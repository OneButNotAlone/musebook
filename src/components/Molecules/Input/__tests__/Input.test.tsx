import * as React from 'react';
import 'jest-styled-components';
import { Checkbox, RadioButton, Dropdown, TextInput } from '../index';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Checkbox', () => {

    test('Checkbox exists', () => {
        expect(Checkbox).toBeDefined();
    });
    
    it('test click event', () => {
        const mockOnClick = jest.fn();

        const checkbox = shallow((<Checkbox label="Option 1" name="option" value="option1" onClick={mockOnClick}/>));
        checkbox.find("input").simulate('click');
        expect(mockOnClick.mock.calls.length).toEqual(1);
    })

    test('Checked checkbox should render correctly', () => {
        const tree = renderer.create(
            <Checkbox label="Option 1" name="option" value="option1" checked={true} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Unchecked checkbox should render correctly', () => {
        const tree = renderer.create(
            <Checkbox label="Option 1" name="option" value="option1" checked={false} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})

test('Checked radio button should render correctly', () => {
    const tree = renderer.create(
        <RadioButton name="option" value="option1" label="Option 1" checked={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Unchecked radio button should render correctly', () => {
    const tree = renderer.create(
        <RadioButton name="option" value="option1" label="Option 1" checked={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Dropdown should render correctly', () => {
    const tree = renderer.create(
        <Dropdown>
            <option>Dropdown</option>
            <option>
                Some crazy long text repeated to make even longer Some crazy long text repeated
                to make even longer Some crazy long text repeated to make even longer
                </option>
        </Dropdown>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Text Input should render correctly', () => {
    const tree = renderer.create(
        <TextInput label="Input Label" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
