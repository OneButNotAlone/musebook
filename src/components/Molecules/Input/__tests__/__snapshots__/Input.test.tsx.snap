// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checked checkbox should render correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

<label
  style={
    Object {
      "color": "#666666",
      "fontFamily": "Bogle, sans-serif",
      "fontSize": "18px",
      "margin": "16px 32px 16px 0",
    }
  }
>
  <input
    checked={true}
    className="c0"
    name="option"
    onClick={[Function]}
    style={
      Object {
        "height": 16,
        "marginRight": 12,
        "maxWidth": 16,
        "minWidth": 16,
      }
    }
    type="checkbox"
    value="option1"
  />
  Option 1
</label>
`;

exports[`Checkbox Unchecked checkbox should render correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

<label
  style={
    Object {
      "color": "#666666",
      "fontFamily": "Bogle, sans-serif",
      "fontSize": "18px",
      "margin": "16px 32px 16px 0",
    }
  }
>
  <input
    checked={false}
    className="c0"
    name="option"
    onClick={[Function]}
    style={
      Object {
        "height": 16,
        "marginRight": 12,
        "maxWidth": 16,
        "minWidth": 16,
      }
    }
    type="checkbox"
    value="option1"
  />
  Option 1
</label>
`;

exports[`Checked radio button should render correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

<label
  style={
    Object {
      "color": "#666666",
      "fontFamily": "Bogle, sans-serif",
      "fontSize": "18px",
      "marginRight": 32,
    }
  }
>
  <input
    checked={true}
    className="c0"
    name="option"
    style={
      Object {
        "height": 16,
        "marginLeft": 1,
        "marginRight": 12,
        "maxWidth": 16,
        "minWidth": 16,
      }
    }
    type="radio"
    value="option1"
  />
  Option 1
</label>
`;

exports[`Dropdown should render correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 36px;
  outline: none;
  border: 1px solid #9b9b9b;
  border-radius: 4px;
  font-family: 'Bogle',sans-serif;
  font-size: 14px;
  color: #666666;
  padding: 0 40px 0 16px;
  background: url('http://www.free-icons-download.net/images/small-down-arrow-icon-15593.png');
  background-repeat: no-repeat;
  background-size: 16px 17px;
  background-position: right 16px center;
}

.c0::-ms-expand {
  display: none;
}

<select
  className="c0 c1"
  type="text"
>
  <option>
    Dropdown
  </option>
  <option>
    Some crazy long text repeated to make even longer Some crazy long text repeated to make even longer Some crazy long text repeated to make even longer
  </option>
</select>
`;

exports[`Text Input should render correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  display: block;
}

.c2 {
  height: 56px;
  outline: none;
  border: 1px solid #9b9b9b;
  border-radius: 4px;
  margin: 24px 0;
  font-family: 'Bogle',sans-serif;
  font-size: 18px;
  padding: 0 16px;
  text-indent: 0;
  color: black;
}

.c2:focus {
  border: 2px solid #2f2951;
  padding: 0 15px;
}

.c2::-webkit-input-placeholder {
  color: #9b9b9b;
}

.c2::-moz-placeholder {
  color: #9b9b9b;
}

.c2:-ms-input-placeholder {
  color: #9b9b9b;
}

.c2::placeholder {
  color: #9b9b9b;
}

<div
  className="c0 c1"
>
  
  <input
    className="c2 c1"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input Label"
    type="text"
  />
</div>
`;

exports[`Unchecked radio button should render correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

<label
  style={
    Object {
      "color": "#666666",
      "fontFamily": "Bogle, sans-serif",
      "fontSize": "18px",
      "marginRight": 32,
    }
  }
>
  <input
    checked={false}
    className="c0"
    name="option"
    style={
      Object {
        "height": 16,
        "marginLeft": 1,
        "marginRight": 12,
        "maxWidth": 16,
        "minWidth": 16,
      }
    }
    type="radio"
    value="option1"
  />
  Option 1
</label>
`;
